pkg/util/strategicpatch/patch.go

```diff
@@ -240,15 +602,20 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	return original, nil
 }
 
-// This panics if any element of the slice is not a map.
+// This method no longer panics if any element of the slice is not a map.
```

Comment:
I wanted to remove all possibility of bad data causing a panic. If this library is linked into the api server, then the cluster could potentially be brought down by submitting bad configuration, unless the panic is removed.